src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -163,20 +229,39 @@ public function finishView(FormView $view, FormInterface $form, array $options)
      */
     public function setDefaultOptions(OptionsResolverInterface $resolver)
     {
-        $choiceListCache = & $this->choiceListCache;
+        $choiceListFactory = $this->choiceListFactory;
+
+        $choiceList = function (Options $options) use ($choiceListFactory) {
+            if (null !== $options['choice_loader']) {
+                // Due to a bug in OptionsResolver, the choices haven't been
+                // validated yet at this point. Remove the if statement once that
+                // bug is resolved
+                if (!$options['choice_loader'] instanceof ChoiceLoaderInterface) {
```

Comment:
This problem in OptionsResolver is fixed with the rewritten implementation since 2.6. So when we set the correct dependency, we can also remove this condition.